src/platform/packages/shared/content-management/table_list_view_table/src/components/use_tag_filter_panel.tsx

```diff
@@ -48,23 +48,12 @@ export const useTagFilterPanel = ({
   addOrRemoveIncludeTagFilter,
 }: Params) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
-  // When the panel is "in use" it means that it is opened and the user is interacting with it.
-  // When the user clicks on a tag to select it, the component is unmounted and mounted immediately, which
-  // creates a new EUI transition "IN" which makes the UI "flicker". To avoid that we pass this
-  // "isInUse" state which disable the transition.
-  const [isInUse, setIsInUse] = useState(false);
   const [options, setOptions] = useState<TagOptionItem[]>([]);
   const [tagSelection, setTagSelection] = useState<TagSelection>({});
   const totalActiveFilters = Object.keys(tagSelection).length;
-  const onClickTime = useRef<number>(0);
 
   const onSelectChange = useCallback(
     (updatedOptions: TagOptionItem[]) => {
-      // onSelectChange() handler is to support keyboard navigation. When user clicks on a tag
-      // we call the onOptionClick() imperatively below and we don't need to do anything here.
-      const timeSinceOptionClick = Date.now() - onClickTime.current;
-      if (timeSinceOptionClick < 100) return;
```

Comment:
This is the main reason for the failure. The failures happened when a test clicked on a tag, but it wasn't selected. This is likely due to the 100 ms timeout, which wasn't enough in a slow testing environment. This 100 ms workaround was meant to skip the default EUI's onSelectionChange logic after we handled the click with our custom onOptionClick handler, but it wasn't enough in a slow test.